src/main/java/com/stripe/net/StripeRequest.java

```diff
@@ -71,7 +71,52 @@ public StripeRequest(
       this.options = (options != null) ? options : RequestOptions.getDefault();
       this.method = method;
       this.url = buildURL(method, url, params);
-      this.content = buildContent(method, params);
+
+      if (this.options instanceof RawRequestOptions) {
+        RawRequestOptions rawOptions = (RawRequestOptions) this.options;
+        this.content = buildContent(method, params, rawOptions.getApiMode());
+      } else {
+        this.content = buildContent(method, params, RawRequestOptions.ApiMode.STANDARD);
+      }
+
+      this.headers = buildHeaders(method, this.options);
+    } catch (IOException e) {
+      throw new ApiConnectionException(
+          String.format(
+              "IOException during API request to Stripe (%s): %s "
+                  + "Please check your internet connection and try again. If this problem persists,"
+                  + "you should check Stripe's service status at https://twitter.com/stripestatus,"
+                  + " or let us know at support@stripe.com.",
+              Stripe.getApiBase(), e.getMessage()),
+          e);
+    }
+  }
+
+  /**
+   * Initializes a new instance of the {@link StripeRequest} class.
+   *
+   * @param method the HTTP method
+   * @param url the URL of the request
+   * @param content the body of the request
+   * @param options the special modifiers of the request
+   * @throws StripeException if the request cannot be initialized for any reason
+   */
+  public StripeRequest(
```

Comment:
Should we create a protected ctor that takes `HttpContent` for content and have public ctors call it? That way we can avoid duplicating all the other logic.

Reply:
Oh yes I like that idea a lot